scripts/githooks/pre-push

```diff
@@ -0,0 +1,7 @@
+#!/bin/bash
+
+mkdir -p build
+cd build
+cmake ..
+make -j`nproc --all`
+ctest
```

Comment:
This seems a little overzealous. For example, I usually run tests in my IDE and don't want to recompile on each push, or I push after a trivial documentation change. I may even want to push to a topic branch despite failing tests (e.g. halfway through implementing a feature).